browser-test/src/program_deep_link.test.ts

```diff
@@ -161,4 +161,27 @@ test.describe('navigating to a deep link', () => {
       await validateScreenshot(page, 'home-page')
     })
   })
+
+  test.describe('with north star flag enabled', () => {
+    test.beforeEach(async ({page}) => {
+      await enableFeatureFlag(page, 'north_star_applicant_ui')
+    })
+
+    test('shows a login prompt for guest users', async ({page}) => {
+      await page.goto('/programs/test-deep-link')
+      await expect(page.getByTestId('slug-login')).toContainText(
+        'Create an account or sign in',
+      )
+      await validateScreenshot(
+        page,
+        'login-prompt-for-guest-users-using-program-slug-north-star',
+      )
```

Comment:
question/suggestion (non-blocking for now, but we'll need to think about this eventually): How does the modal do with a11y checks? Accessibility has been an issue with modals in the past and I think the hope is moving to thymeleaf will help with that. Do the tests pass if we add a call to `validateAccessibility`?

Reply:
added validateAccessibility and it passed. The USWDS modals I think should be an improvement